Automattic/wp-calypso
# format-currency: Show `US$` for USD outside of US

In the US, users will expect to see USD prices rendered with the currency symbol `$`. However, there are many other currencies which use `$` as their currency symbol (eg: `CAD`). The `@automattic/format-currency` functions try to prevent confusion between these symbols by using an international version of the symbol when the locale does not match the currency. So if your locale is `en-CA`, USD prices will be rendered with the symbol `US$`. 

However, this relies on the user having set their interface language to something other than `en-US`/`en`, and many English-speaking non-US users still have that interface language (eg: there's no English locale available in our settings for Argentinian English so such users would probably still have `en`). As a result, those users will see a price with `$` and could be misled about what currency is being displayed.

## Proposed Changes

In this diff, we fetch the user's geolocation (using the `/geo` API endpoint which does not require authentication) and use it to override the currency symbol for USD. If the user is in the US, they will see USD with the `$` symbol, otherwise they will see the `US$` symbol.

The fetching should occur as early as possible during calypso's loading process and should be attempted for both logged-in and logged-out users (eg: so that the logged-out version of the Plans page shows the correct currency symbol). To do this, we make the API call inside the "layout" component. The call is a non-blocking side effect so it's entirely possible for it to fail without any harmful effects.

Fixes https://github.com/Automattic/wp-calypso/issues/73496

## Screenshots

Inside US:

<img width="568" alt="Screenshot 2023-03-10 at 4 26 09 PM" src="https://user-images.githubusercontent.com/2036909/224433970-bdae0da6-a250-44e1-874e-0fcb83c02b1c.png">

Outside US:

<img width="572" alt="Screenshot 2023-03-10 at 4 26 20 PM" src="https://user-images.githubusercontent.com/2036909/224433983-8537d8ca-d01a-472f-a6e1-74887ff7e1a1.png">

## Testing Instructions

- View calypso with a user that has `en` or `en-US` as their locale and `USD` as their currency.
- Visit a page in calypso that has at least one price displayed (eg: `/plans` or `/checkout`).
- Make your geolocation inside the US (using a VPN if necessary) and verify that the prices are displayed with `$`.
- Make your geolocation outside the US (using a VPN if necessary) and verify that the prices are displayed with `US$`.
